### PR TITLE
Add tests on variable.py

### PR DESCRIPTION
**Pull request by raphaelauv**
- add a test on the order of the check of the secret_backends
- few fix on code
- Add a test to check if loading LocalFilesystemBackend as a secret_backend work

### CODE REVIEW

File: airflow/models/variable.py [hunk 1]
```diff
@@ -223,6 +223,7 @@ def rotate_fernet_key(self):
         if self._val and self.is_encrypted:
             self._val = fernet.rotate(self._val.encode('utf-8')).decode()
 
+    @staticmethod
```
Comment (ashb) [inline airflow/models/variable.py:226]: Wat. How did this ever work without this?

Comment (potiuk) [reply airflow/models/variable.py:226]: Ough. Maybe that's why I was under the impression it wasn't working before - becuase it got the "class"  object (?) passed as key !

**Review by kaxil — COMMENTED** [review]
Can you fix the static check failure please: https://github.com/apache/airflow/runs/4096970541?check_suite_focus=true

File: airflow/models/variable.py (hunk 1 above)
Comment (potiuk) [reply airflow/models/variable.py:226]: Though I also do not understand it :)

**Comment by raphaelauv** [comment]
@kaxil should I change the base branch to main ?

```
Update version to the latest version in the documentation................................Failed
- hook id: update-version
- files were modified by this hook

Replacing re.compile('(FROM apache/airflow:).*($)') to 2.2.1 in /home/runner/work/airflow/airflow/docs/docker-stack/docker-examples/extending/writable-directory/Dockerfile
Replacing re.compile('(FROM apache/airflow:).*($)') to 2.2.1 in /home/runner/work/airflow/airflow/docs/docker-stack/docker-examples/extending/embedding-dags/Dockerfile
Replacing re.compile('(FROM apache/airflow:).*($)') to 2.2.1 in /home/runner/work/airflow/airflow/docs/docker-stack/docker-examples/extending/add-pypi-packages/Dockerfile
Replacing re.compile('(FROM apache/airflow:).*($)') to 2.2.1 in /home/runner/work/airflow/airflow/docs/docker-stack/docker-examples/extending/add-build-essential-extend/Dockerfile
Replacing re.compile('(FROM apache/airflow:).*($)') to 2.2.1 in /home/runner/work/airflow/airflow/docs/docker-stack/docker-examples/extending/add-providers/Dockerfile
Replacing re.compile('(FROM apache/airflow:).*($)') to 2.2.1 in /home/runner/work/airflow/airflow/docs/docker-stack/docker-examples/extending/add-apt-packages/Dockerfile
Replacing re.compile('(apache/airflow:)[^-]*(\\-)') to 2.2.1 in /home/runner/work/airflow/airflow/docs/docker-stack/entrypoint.rst
Replacing re.compile('(/constraints-)[^-]*(/constraints)') to 2.2.1 in /home/runner/work/airflow/airflow/docs/docker-stack/docker-examples/restricted/restricted_environments.sh
Replacing re.compile('(AIRFLOW_VERSION=")[^"]*(" \\\\)') to 2.2.1 in /home/runner/work/airflow/airflow/docs/docker-stack/docker-examples/restricted/restricted_environments.sh
```

**Comment by kaxil** [comment]
Oh I hadn't noticed you are not on the "main" branch. Yes please can you rebase on the main branch

**Comment by raphaelauv** [comment]
@kaxil done , thanks for your help

**Review by kaxil — APPROVED** [review]

**Review by jedcunningham — APPROVED** [review]

**Comment by kaxil** [comment]
All the failed tests have "137" -- either that is a real bug introduced in "main" where some test or code is leaking memory or some wrong with runners.

**Pull request merged**